spec/lita/handlers/api/market_spec.rb

```diff
@@ -0,0 +1,188 @@
+require 'spec_helper'
+
+describe Lita::Handlers::Api::Market, lita_handler: true do
+  let(:redis) { Lita::Handlers::LunchReminder.new(robot).redis }
+  let(:karmanager) { Lita::Services::Karmanager.new(redis) }
+  let(:assigner) { Lita::Services::LunchAssigner.new(redis, karmanager) }
+  let(:market) { Lita::Services::MarketManager.new(redis, assigner, karmanager) }
```

Comment:
Como cada servicio tiene su test (o debería) en vez de crearlos deberías ocupar mocks. Como la API es solamente un controlador, solo deberías preocuparte que llame a los métodos indicados de los servicios y devuelva la respuesta correcta en cada caso.

En resumen, podrías cambiar `let(:karmanager) { Lita::Services::Karmanager.new(redis) }` por `let(:karmanager) { double }`. Y hacer los `allow(karmanager).to receive(:method_X).and_return(Y)` necesarios para que funcione.

Así te ahorras definir la variable `redis`, que no tiene relación con la API (porque no ocupa redis directamente).